syntax = "proto3";

package proto_test;

message ResultCode{
  int32 errcode = 1;
  string errmsg = 2;
};

enum Gender{
  GENDER_UNSPECIFIED = 0;
  GENDER_MALE = 1;
  GENDER_FEMALE = 2;
};

message User{ 
  string name = 1;
  uint32 age = 2;
  Gender gender = 3;  
};

message LoginRequest{
  string name = 1;
  string password  = 2;
};

message LoginResponse{
  ResultCode result_code = 1;
  bool success = 2;
};

message GetFriendListRequest{
  uint32 userid = 1;
};

message GetFriendListResponse{
  ResultCode result_code = 1;
  repeated User friend_list = 2; 
};

/// 用户服务，提供登录和好友列表查询等远程调用接口
service UserServiceRpc{
  /// 用户登录验证
  /// @param request 登录请求，包含用户名和密码 
  /// @return 登录响应，包含结果码和成功状态
  rpc Login(LoginRequest)returns(LoginResponse);

  /// 获取指定用户的好友列表
  /// @param request 请求，包含用户ID
  /// @return 响应，包含结果码和好友列表
  rpc GetFriendList(GetFriendListRequest)returns(GetFriendListResponse);
};